extensions/dataformat/runtime/src/main/java/org/apache/camel/quarkus/component/dataformat/CamelDataformatConfig.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.dataformat;
+
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithParentName;
+
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+@ConfigMapping(prefix = "camel.dataformat")
+public interface CamelDataformatConfig {
+
+    /**
+     * Configuration of beanio dataformat.
+     *
+     * @asciidoclet
+     */
+    Optional<DataformatConfig> beanio();
+
+    /**
+     * Configuration of properties for any dataformat.
+     * (In the format camel.dataformat."dataformat_name"."property"=value
+     *
+     * i.e.
+     * camel.dataformat.beanio.stream-name = test-stream
+     * camel.dataformat.beanio.mapping = test-mapping.xml
+     */
+    @WithParentName
+    Map<String, Map<String, String>> dataformats();
+
+    /**
+     * Configuration of beanio dataformat.
+     */
+    interface DataformatConfig {
```

Comment:
I think you can probably remove this and the `beanio()` config. It's only for docs, right? It's already covered by the Camel component docs.

Reply:
you are right, the docs is the only befit. I'll remove it

Reply:
fixed